entities/container.py

```diff
@@ -12,6 +17,63 @@ def __init__(self, env, region):
 
         self.region = region
         self.location = region.draw_location()
+        self.id = next(self._ids)
+        self.state = ContainerState.EMPTY
+
+
+    def create_bids(self, best_shipments, registrationkey):
+        containerbid = namedtuple('containerbid', 'container_registration_key shipment_registration_key biddingvalue')
+        containerbids = []
+        for entry in best_shipments:
+            shipment_registration_key = entry[0]
+            shipment = entry[2]
+            transport_cost_empty = route_euclidean_distance(self.env, self.location, shipment.location)
+            transport_cost_shipment = route_euclidean_distance(self.env, shipment.location, shipment.destination)
+            biddingvalue = transport_cost_empty + transport_cost_shipment
+            #TODO add additional profit margin for container or decrease of biddingvalue based on urgency
+            container_bid = containerbid(container_registration_key = registrationkey,
+                                        shipment_registration_key = shipment_registration_key,
+                                        biddingvalue = biddingvalue)
+            containerbids.append(container_bid)
+        return containerbids
+
+    def request_shipments(self):
+        #TODO add functionality of selecting shipments from auctions outside current region
+        if EntityTypes.__members__['SHIPMENT'] not in \
+                self.region.auctioneer.entities.keys(): # There are currently no shipments available
+            return
+        else:
+            available_shipments = self.region.auctioneer.entities[EntityTypes.SHIPMENT]
+        return available_shipments # dict with shipment objects and their registration keys
+
+    def select_best_shipments(self, available_shipments):
+        ''' In the final model, the container should make transport reservations for each shipment it bids on,
+        therefore the container places a limited number of bids. The container makes a selection based on the
+        distance to the shipments. The closer a container is to a shipment, the lower it can bid.'''
+        def take_distance(elem):
+            return elem[1]
+
+        distance_to_shipments = [] # change to namedtuple to provide better insight in bid function?
+
+        for key in available_shipments.keys():
+            shipment = self.region.auctioneer.entities[EntityTypes.SHIPMENT][key]
+            distance = route_euclidean_distance(self.env, self.location, shipment.location)
+            registrationkey = key
+            distance_to_shipments.append([registrationkey, distance, shipment])
+
+        distance_to_shipments.sort(key = take_distance)
```

Comment:
No need for a `def` above. Use `key=lambda item: item[1]` instead. (Lambdas are anonymous functions).